### PR TITLE
Testing Files and CMakeLists.txt

### LINKED ISSUE #24: Unit Test Cases for Airspeed Data Extraction
Given the airspeed data extraction logic is implemented
When the plugin is executed and unit tests are run
Then the airspeed extraction functionality is validated through unit tests

Developer Notes:
Write unit test cases to verify the correct extraction of the airspeed DataRef from X-Plane. This ensures that the data extraction logic works as expected and helps to prevent regressions in future development.

Acceptance Criteria:

Unit test cases are written for the airspeed extraction logic.
Tests are run and pass without errors.
The test framework is set up and integrated into the project.

Testing Notes:

Run the unit tests using the defined test framework (e.g., Google Test or Catch2).
Verify that all test cases pass and the airspeed data is correctly extracted.

### PR DESCRIPTION
**Pull request by carlyrhoover**
X
x

### Fixes #24 

---

### **What was changed?**
-added a 'CMakeLists.txt' as well as a header file and the 'plugin_tests.cpp' file within the tests subdirectory for the proper testing build using CMake
---

### **Why was it changed?**

This sets up the correct structure for testing

---

### **How was it changed?**
 Main plugin file was not changed, more tests need to be added to correctly accomodate the plugin file 

---

### CODE REVIEW

File: include/plugin.h
Comment (aelnourSLU) [inline include/plugin.h:1]: No objections, but for my edification, what is the intent with externalizing these functions for the plugin? Will this be used for the TCP communication to iMotions?

File: tests/plugin_tests.cpp
Comment (aelnourSLU) [inline tests/plugin_tests.cpp:1]: If possible, I'd like to apply the same formatting rules for our main cpp file here, using the recommended C/C++ formatter.

File: tests/plugin_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include <gtest/gtest.h>
+#include "plugin.h"  // You'll need to create this header file
```
Comment (aelnourSLU) [inline tests/plugin_tests.cpp:2]: What is this comment in reference to precisely?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,43 @@
+cmake_minimum_required(VERSION 3.10)
+project(XPlanePlugin)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# X-Plane SDK path (you'll need to adjust this)
```
Comment (aelnourSLU) [inline CMakeLists.txt:7]: The relative path for the SDK is now standardized by the team, we should have no need to adjust this

**Review by aelnourSLU — APPROVED** [review]

**Comment by hrmeiners** [comment]
I like the framework you've got here with Google Tests, but it's hard to say what to change or do next without the standardized testing example from @harshitha-thota. Once we get that example code, we should be able to more focus on the exact specifics of what we are supposed to accomplish. Once that confusion clears up and both of us align our code to those specific goals, we should have a better idea of what we need to place here!

**Pull request closed without merging**